shared/metrics/src/main/java/io/pravega/shared/MetricsNames.java

```diff
@@ -240,13 +240,13 @@ public static String failMetricName(String metricName) {
     }
 
     /**
-     * Convenient method to concastenate two Strings using dot.
+     * Convenient method to join two Strings with dot.
      *
      * @param element1 the first String object.
      * @param element2 the second String object.
-     * @return String object with element1 and element2 connected with dot.
+     * @return String object with element1 and element2 joined with dot.
      */
-    public static String name(String element1, String element2) {
+    public static String joinWithDot(String element1, String element2) {
```

Comment:
In all the public methods of this class we are using names that refer to the metrics, right? Instead, the name used for this method exposes why it does. Similarly to the rest of public methods, I think that we need to change it to something like `buildMetricName`, as it describe "what" it does (how it does it (using a `.` or a `-`), is an implementation detail).

Reply:
This method is there simply because one static method "name" was gone with Dropwizard library.
"buildMetricName" is not a good name either as this method doesn't always return metric name. (metric name is quite complex so we even have MetricKey class and metricKey method to handle it). It indeed just joins two string with dot.

If you still insist a name change, could you advise one here and let's agree on it before I commit code change.